Improve opaque type higher-ranked region error message under NLL

Aaron1011 opened the pull request:
Currently, any higher-ranked region errors involving opaque types
fall back to a generic "higher-ranked subtype error" message when
run under NLL. This PR adds better error message handling for this
case, giving us the same kinds of error messages that we currently
get without NLL:

```
error: implementation of `MyTrait` is not general enough
  --> $DIR/opaque-hrtb.rs:12:13
   |
LL | fn foo() -> impl for<'a> MyTrait<&'a str> {
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ implementation of `MyTrait` is not general enough
   |
   = note: `impl MyTrait<&'2 str>` must implement `MyTrait<&'1 str>`, for any lifetime `'1`...
   = note: ...but it actually implements `MyTrait<&'2 str>`, for some specific lifetime `'2`

error: aborting due to previous error
```

To accomplish this, several different refactoring needed to be made:

* We now have a dedicated `InstantiateOpaqueType` struct which
implements `TypeOp`. This is used to invoke `instantiate_opaque_types`
during MIR type checking.
* `TypeOp` is refactored to pass around a `MirBorrowckCtxt`, which is
needed to report opaque type region errors.
* We no longer assume that all `TypeOp`s correspond to canonicalized
queries. This allows us to properly handle opaque type instantiation
(which does not occur in a query) as a `TypeOp`.
A new `ErrorInfo` associated type is used to determine what
additional information is used during higher-ranked region error
handling.
* The body of `try_extract_error_from_fulfill_cx`
has been moved out to a new function `try_extract_error_from_region_constraints`.
This allows us to re-use the same error reporting code between
canonicalized queries (which can extract region constraints directly
from a fresh `InferCtxt`) and opaque type handling (which needs to take
region constraints from the pre-existing `InferCtxt` that we use
throughout MIR borrow checking).

rust-highfive commented:
r? @davidtwco

(rust-highfive has picked a reviewer for you, use r? to override)

Aaron1011 commented:
cc @matthewjasper @oli-obk @lqd

Aaron1011 commented on src/test/ui/nll/relate_tys/opaque-hrtb.stderr:
```diff
@@ -0,0 +1,11 @@
+  --> $DIR/opaque-hrtb.rs:12:13
+   |
+LL | fn foo() -> impl for<'a> MyTrait<&'a str> {
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ implementation of `MyTrait` is not general enough
```
This span isn't great, but it's the same one that we get without `#![feature(nll)]`. I'll probably work on improving it in a follow-up PR.

oli-obk replied:
Most such spans will get better with my lazy TAIT PR

Aaron1011 commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion.

@rustbot label: +S-waiting-on-perf

bors commented:
:hourglass: Trying commit 1bd8e29ed18ad76b00d71d3d51b1d5a94888e6d4 with merge 9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67...
<!-- homu: {"type":"TryBuildStarted","head_sha":"1bd8e29ed18ad76b00d71d3d51b1d5a94888e6d4","merge_sha":"9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/4639305179?check_suite_focus=true)
Build commit: 9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67 (`9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/4639305179?check_suite_focus=true"},"merge_sha":"9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67"} -->

rust-timer commented:
Queued 9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67 with parent f8abed9ed48bace6be0087bcd44ed534e239b8d8, future [comparison URL](https://perf.rust-lang.org/compare.html?start=f8abed9ed48bace6be0087bcd44ed534e239b8d8&end=9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67).

rust-timer commented:
Finished benchmarking commit (9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67): [comparison url](https://perf.rust-lang.org/compare.html?start=f8abed9ed48bace6be0087bcd44ed534e239b8d8&end=9cb40efda24c4b68d6c59a8e3d6d294ac2f86e67).

**Summary**: This change led to moderate relevant regressions 😿 in compiler performance.
- Moderate regression in instruction counts (up to 1.2% on `incr-unchanged` builds of `deeply-nested-async`)

If you disagree with this performance assessment, please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. While you can manually mark this PR as fit for rollup, we strongly recommend not doing so since this PR led to changes in compiler perf.

**Next Steps**: If you can justify the regressions found in this try perf run, please indicate this with `@rustbot label: +perf-regression-triaged` along with sufficient written justification. If you cannot justify the regressions please fix the regressions and do another perf run. If the next run shows neutral or positive results, the label will be automatically removed.

@bors rollup=never
@rustbot label: +S-waiting-on-review -S-waiting-on-perf +perf-regression

oli-obk reviewed (commented):
This is really cool and may make some things easier for me actually

oli-obk commented:
r? @oli-obk

bors commented:
:umbrella: The latest upstream changes (presumably #92844) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:umbrella: The latest upstream changes (presumably #92007) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

Aaron1011 commented:
@oli-obk I've rebased this now that https://github.com/rust-lang/rust/pull/92007 has been merged. The PR is pretty much the same - I just changed the new type-op to work with the new `opaque_ty_obligation` method.

oli-obk commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion.

@rustbot label: +S-waiting-on-perf

bors commented:
:hourglass: Trying commit 880d8e2a06157563f1fe47fb5125a96e7037fced with merge 0fd10de714ecd9c91b80fb63c4fffd13e1f68f56...
<!-- homu: {"type":"TryBuildStarted","head_sha":"880d8e2a06157563f1fe47fb5125a96e7037fced","merge_sha":"0fd10de714ecd9c91b80fb63c4fffd13e1f68f56"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/5107465323?check_suite_focus=true)
Build commit: 0fd10de714ecd9c91b80fb63c4fffd13e1f68f56 (`0fd10de714ecd9c91b80fb63c4fffd13e1f68f56`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/5107465323?check_suite_focus=true"},"merge_sha":"0fd10de714ecd9c91b80fb63c4fffd13e1f68f56"} -->

rust-timer commented:
Queued 0fd10de714ecd9c91b80fb63c4fffd13e1f68f56 with parent 2a8dbdb1e2d9f44e7971d9a70b41d4d86db0112f, future [comparison URL](https://perf.rust-lang.org/compare.html?start=2a8dbdb1e2d9f44e7971d9a70b41d4d86db0112f&end=0fd10de714ecd9c91b80fb63c4fffd13e1f68f56).

rust-timer commented:
Finished benchmarking commit (0fd10de714ecd9c91b80fb63c4fffd13e1f68f56): [comparison url](https://perf.rust-lang.org/compare.html?start=2a8dbdb1e2d9f44e7971d9a70b41d4d86db0112f&end=0fd10de714ecd9c91b80fb63c4fffd13e1f68f56).

**Summary**: This benchmark run did not return any relevant results.

If you disagree with this performance assessment, please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. While you can manually mark this PR as fit for rollup, we strongly recommend not doing so since this PR led to changes in compiler perf.

@bors rollup=never
@rustbot label: +S-waiting-on-review -S-waiting-on-perf -perf-regression

oli-obk commented:
@bors r+

Awesome!

bors commented:
:pushpin: Commit 880d8e2a06157563f1fe47fb5125a96e7037fced has been approved by `oli-obk`

<!-- @bors r=oli-obk 880d8e2a06157563f1fe47fb5125a96e7037fced -->
<!-- homu: {"type":"Approved","sha":"880d8e2a06157563f1fe47fb5125a96e7037fced","approver":"oli-obk"} -->

Aaron1011 commented:
@bors r=oli-obk

bors commented:
:pushpin: Commit 48a48fd1b85ce50a890303b7238550ccf0613433 has been approved by `oli-obk`

<!-- @bors r=oli-obk 48a48fd1b85ce50a890303b7238550ccf0613433 -->
<!-- homu: {"type":"Approved","sha":"48a48fd1b85ce50a890303b7238550ccf0613433","approver":"oli-obk"} -->

bors commented:
:hourglass: Testing commit 48a48fd1b85ce50a890303b7238550ccf0613433 with merge 27ee5ca00e0d78f12757b37a3a8592495491c1f2...
<!-- homu: {"type":"BuildStarted","head_sha":"48a48fd1b85ce50a890303b7238550ccf0613433","merge_sha":"27ee5ca00e0d78f12757b37a3a8592495491c1f2"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/5115893083?check_suite_focus=true) [(plain)](https://github.com/rust-lang-ci/rust/commit/27ee5ca00e0d78f12757b37a3a8592495491c1f2/checks/5115893083/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] ui/nll/relate_tys/fn-subtype.rs ... ok
test [ui] ui/nll/relate_tys/hr-fn-aba-as-aaa.rs ... ok
test [ui] ui/nll/relate_tys/impl-fn-ignore-binder-via-bottom.rs ... ok
test [ui] ui/nll/relate_tys/issue-48071.rs ... ok
test [ui] ui/nll/relate_tys/opaque-hrtb.rs ... ok
test [ui] ui/nll/return-ref-mut-issue-46557.rs ... ok
test [ui] ui/nll/ty-outlives/impl-trait-captures.rs ... ok
test [ui] ui/nll/return_from_loop.rs ... ok
test [ui] ui/nll/relate_tys/trait-hrtb.rs ... ok
---

---- [ui] ui/cmse-nonsecure/cmse-nonsecure-call/params-on-stack.rs stdout ----
diff of stderr:

1 error: <unknown>:0:0: in function test i32 (i32, i32, i32, i32, i32): call to non-secure function would require passing arguments on stack
+ thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
+   left: `LLVMing`,
+   left: `LLVMing`,
+  right: `Codegenning`', /checkout/compiler/rustc_codegen_ssa/src/back/write.rs:1470:21
+    0:     0x7feb3677b85c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h10995e31cbc01166
+    1:     0x7feb367e9cce - core::fmt::write::hd7d6faa80ce2553b
+    1:     0x7feb367e9cce - core::fmt::write::hd7d6faa80ce2553b
+    2:     0x7feb3676a771 - std::io::Write::write_fmt::h3f5abf7b9f1eca07
+    3:     0x7feb3677b68b - std::sys_common::backtrace::print::h4d653406e7d402ad
+    4:     0x7feb3677fea4 - std::panicking::default_hook::{{closure}}::h73e6d85c2226b99a
+    5:     0x7feb3677fa86 - std::panicking::default_hook::h578dc98dd31a3735
+    6:     0x7feb378f087a - rustc_driver[52bb940b4ffe0591]::DEFAULT_HOOK::{closure#0}::{closure#0}
+    7:     0x7feb367805c3 - std::panicking::rust_panic_with_hook::hf831cb87f6160c47
+    8:     0x7feb367803d7 - std::panicking::begin_panic_handler::{{closure}}::h3962d95fa5d93909
+    9:     0x7feb3677bd74 - std::sys_common::backtrace::__rust_end_short_backtrace::hbbae4df880a75e04
+   10:     0x7feb36780099 - rust_begin_unwind
+   11:     0x7feb36736a53 - core::panicking::panic_fmt::h2cba5c8061a950b9
+   12:     0x7feb367e6658 - core::panicking::assert_failed_inner::h19ad8376367d71e3
+   13:     0x7feb373fb93b - core[ce52307a21260d31]::panicking::assert_failed::<rustc_codegen_ssa[f59136af200a9e6b]::back::write::MainThreadWorkerState, rustc_codegen_ssa[f59136af200a9e6b]::back::write::MainThreadWorkerState>
+   14:     0x7feb37c7df5e - rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work::<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}
+   15:     0x7feb37c9a646 - std[962aeb78bac5d12c]::sys_common::backtrace::__rust_begin_short_backtrace::<<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend as rustc_codegen_ssa[f59136af200a9e6b]::traits::backend::ExtraBackendMethods>::spawn_thread<rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#0}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>
+   16:     0x7feb37c77a39 - <<std[962aeb78bac5d12c]::thread::Builder>::spawn_unchecked_<<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend as rustc_codegen_ssa[f59136af200a9e6b]::traits::backend::ExtraBackendMethods>::spawn_thread<rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#0}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#1} as core[ce52307a21260d31]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
+   17:     0x7feb3678f3a3 - std::sys::unix::thread::Thread::new::thread_start::h631d624327c9afc0
+   18:     0x7feb364b8609 - start_thread
+   19:     0x7feb365fa293 - clone
+   20:                0x0 - <unknown>
+ error: internal compiler error: unexpected panic
+ 
+ note: the compiler unexpectedly panicked. this is a bug.
+ 
+ 
+ note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md
+ 
+ note: rustc 1.60.0-nightly (27ee5ca00 2022-02-08) running on x86_64-unknown-linux-gnu
+ 
+ note: compiler flags: -Z threads=1 -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C prefer-dynamic -C rpath -C debuginfo=0 --crate-type lib
+ query stack during panic:
+ end of query stack
3 error: aborting due to previous error
4 
---
To only update this specific test, also pass `--test-args cmse-nonsecure/cmse-nonsecure-call/params-on-stack.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/cmse-nonsecure/cmse-nonsecure-call/params-on-stack.rs" "-Zthreads=1" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/cmse-nonsecure/cmse-nonsecure-call/params-on-stack" "-A" "unused" "-Crpath" "-O" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target" "thumbv8m.main-none-eabi" "--crate-type" "lib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/cmse-nonsecure/cmse-nonsecure-call/params-on-stack/auxiliary"
------------------------------------------
Some tests failed in compiletest suite=ui mode=ui host=x86_64-unknown-linux-gnu target=x86_64-unknown-linux-gnu

------------------------------------------
------------------------------------------
stderr:
------------------------------------------
error: <unknown>:0:0: in function test i32 (i32, i32, i32, i32, i32): call to non-secure function would require passing arguments on stack
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `LLVMing`,
  left: `LLVMing`,
 right: `Codegenning`', /checkout/compiler/rustc_codegen_ssa/src/back/write.rs:1470:21
   0:     0x7feb3677b85c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h10995e31cbc01166
   1:     0x7feb367e9cce - core::fmt::write::hd7d6faa80ce2553b
   1:     0x7feb367e9cce - core::fmt::write::hd7d6faa80ce2553b
   2:     0x7feb3676a771 - std::io::Write::write_fmt::h3f5abf7b9f1eca07
   3:     0x7feb3677b68b - std::sys_common::backtrace::print::h4d653406e7d402ad
   4:     0x7feb3677fea4 - std::panicking::default_hook::{{closure}}::h73e6d85c2226b99a
   5:     0x7feb3677fa86 - std::panicking::default_hook::h578dc98dd31a3735
   6:     0x7feb378f087a - rustc_driver[52bb940b4ffe0591]::DEFAULT_HOOK::{closure#0}::{closure#0}
   7:     0x7feb367805c3 - std::panicking::rust_panic_with_hook::hf831cb87f6160c47
   8:     0x7feb367803d7 - std::panicking::begin_panic_handler::{{closure}}::h3962d95fa5d93909
   9:     0x7feb3677bd74 - std::sys_common::backtrace::__rust_end_short_backtrace::hbbae4df880a75e04
  10:     0x7feb36780099 - rust_begin_unwind
  11:     0x7feb36736a53 - core::panicking::panic_fmt::h2cba5c8061a950b9
  12:     0x7feb367e6658 - core::panicking::assert_failed_inner::h19ad8376367d71e3
  13:     0x7feb373fb93b - core[ce52307a21260d31]::panicking::assert_failed::<rustc_codegen_ssa[f59136af200a9e6b]::back::write::MainThreadWorkerState, rustc_codegen_ssa[f59136af200a9e6b]::back::write::MainThreadWorkerState>
  14:     0x7feb37c7df5e - rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work::<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}
  15:     0x7feb37c9a646 - std[962aeb78bac5d12c]::sys_common::backtrace::__rust_begin_short_backtrace::<<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend as rustc_codegen_ssa[f59136af200a9e6b]::traits::backend::ExtraBackendMethods>::spawn_thread<rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#0}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>
  16:     0x7feb37c77a39 - <<std[962aeb78bac5d12c]::thread::Builder>::spawn_unchecked_<<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend as rustc_codegen_ssa[f59136af200a9e6b]::traits::backend::ExtraBackendMethods>::spawn_thread<rustc_codegen_ssa[f59136af200a9e6b]::back::write::start_executing_work<rustc_codegen_llvm[96c54cfd82f181b2]::LlvmCodegenBackend>::{closure#4}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#0}, core[ce52307a21260d31]::result::Result<rustc_codegen_ssa[f59136af200a9e6b]::back::write::CompiledModules, ()>>::{closure#1} as core[ce52307a21260d31]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  17:     0x7feb3678f3a3 - std::sys::unix::thread::Thread::new::thread_start::h631d624327c9afc0
  18:     0x7feb364b8609 - start_thread
  19:     0x7feb365fa293 - clone
  20:                0x0 - <unknown>
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.60.0-nightly (27ee5ca00 2022-02-08) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C prefer-dynamic -C rpath -C debuginfo=0 --crate-type lib
query stack during panic:
end of query stack
error: aborting due to previous error

```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/5116209287?check_suite_focus=true)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/5116209287?check_suite_focus=true","builder_name":"checks-actions"} -->

Aaron1011 commented:
@bors retry hit spurious failure https://github.com/rust-lang/rust/pull/92306

bors commented:
:hourglass: Testing commit 48a48fd1b85ce50a890303b7238550ccf0613433 with merge 1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049...
<!-- homu: {"type":"BuildStarted","head_sha":"48a48fd1b85ce50a890303b7238550ccf0613433","merge_sha":"1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/5125327495?check_suite_focus=true)
Approved by: oli-obk
Pushing 1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"oli-obk","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/5125327495?check_suite_focus=true"},"merge_sha":"1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049"} -->

rust-timer commented:
Finished benchmarking commit (1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049): [comparison url](https://perf.rust-lang.org/compare.html?start=bf242bb1199e25ca2274df5c4114e0c9436b74e9&end=1f0a96862ac9d4c6ca3e4bb500c8b9eac4d83049).

**Summary**: This benchmark run did not return any relevant results.

If you disagree with this performance assessment, please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).



@rustbot label: -perf-regression
